### DISCUSSION TITLE
ELI5 FloatingTree in Dropdown Menu

### DISCUSSION THREAD

**Discussion by rthai** [floating-ui/floating-ui#1750] [Q&A]
Hi 👋. I'm not quite understanding the Dropdown Menu example and how it uses `<FloatingTree />`. My understanding of `FloatingTree` is to use it when I want to support nested floating elements. I don't want to support submenus in the Menu component I'm writing, so why do I still need `FloatingTree`?

In the [Dropdown Menu codesandbox example](https://codesandbox.io/s/admiring-lamport-5wt3yg?file=/src/DropdownMenu.tsx), why does the `<MenuComponent />` need to be wrapped in a `<FloatingTree />` in the following manner?  

```jsx
export const Menu: React.FC<Props> = forwardRef((props, ref) => {
  const parentId = useFloatingParentNodeId();

  if (parentId == null) {
    return (
      <FloatingTree>
        <MenuComponent {...props} ref={ref} />
      </FloatingTree>
    );
  }

  return <MenuComponent {...props} ref={ref} />;
});
```

Vs. Using `FloatingTree` in `<MenuComponent />` like below? When I do this, `tree` (ln#68 in codesandbox example) is undefined. 

```jsx
export const MenuComponent = forwardRef<
  any,
  Props & React.HTMLProps<HTMLButtonElement>
>(({ children, label, ...props }, ref) => {

 ...

  return (
    <FloatingTree>
      <FloatingNode id={nodeId}>
        {/* trigger element */}
        <FloatingPortal>
          {/* dropdown stuff */}
        </FloatingPortal>
      </FloatingNode>
    <FloatingTree/>
  );
});
```

Thanks!

**Comment by atomiks** [comment]
You don't need `FloatingTree` if you don't have submenus. 

The API allows you to cleanly nest `Menu` inside another `Menu` as a child in place of a `MenuItem`, but this has no relevance if you aren't nesting.

```js
<Menu label="Edit">
  <MenuItem label="Undo" />
  <MenuItem label="Redo" />
  <Menu label="Copy as">
    <MenuItem label="Text" />
    <MenuItem label="Video" />
    <Menu label="Image">
      <MenuItem label=".png" />
      <MenuItem label=".jpg" />
      <MenuItem label=".svg" />
      <MenuItem label=".gif" />
    </Menu>
    <MenuItem label="Audio" />
  </Menu>
</Menu>
```

`<FloatingTree>` needs to wrap all of these nodes at the top-level, in your example you're creating a new tree for each node
